tests/Dfc.CourseDirectory.Core.Tests/BackgroundWorkersTests/QueueBackgroundWorkSchedulerTests.cs

```diff
@@ -0,0 +1,170 @@
+            {
+                executedSignal.Set();
+                return Task.CompletedTask;
+            };
+
+            // Act
+            await scheduler.StartAsync(cancellationToken: default);
+            await scheduler.Schedule(workItem);
+
+            // Assert
+            executedSignal.Wait(100);
+        }
+
+        [Fact]
+        public async Task DoesNotRethrowExceptionFromWorkItem()
+        {
+            // Arrange
+            var serviceScopeFactory = CreateServiceScopeFactory();
+            var loggerFactory = CreateLoggerFactory();
+            using var scheduler = new QueueBackgroundWorkScheduler(serviceScopeFactory, loggerFactory);
+
+            using var executedSignal = new ManualResetEventSlim(false);
+            WorkItem workItem = (state, sp, ct) =>
+            {
+                try
+                {
+                    throw new Exception("Bang!");
+                }
+                finally
+                {
+                    executedSignal.Set();
+                }
+            };
+
+            // Act
+            await scheduler.StartAsync(cancellationToken: default);
+            await scheduler.Schedule(workItem);
+            executedSignal.Wait(100);
+
+            // Assert
+            await scheduler.StopAsync(cancellationToken: default);
+        }
+
+        [Fact]
+        public async Task PassesStateToWorkItem()
+        {
+            // Arrange
+            var serviceScopeFactory = CreateServiceScopeFactory();
+            var loggerFactory = CreateLoggerFactory();
+            using var scheduler = new QueueBackgroundWorkScheduler(serviceScopeFactory, loggerFactory);
+
+            var workItemState = new object();
+
+            using var executedSignal = new ManualResetEventSlim(false);
+            WorkItem workItem = (state, sp, ct) =>
+            {
+                Assert.Same(workItemState, state);
+
+                executedSignal.Set();
+                return Task.CompletedTask;
+            };
+
+            // Act
+            await scheduler.StartAsync(cancellationToken: default);
+            await scheduler.Schedule(workItem, workItemState);
+            executedSignal.Wait(100);
+        }
+
+        [Fact]
+        public async Task PassesCancellationTokenToWorkItem()
+        {
+            // Arrange
+            var serviceScopeFactory = CreateServiceScopeFactory();
+            var loggerFactory = CreateLoggerFactory();
+            using var scheduler = new QueueBackgroundWorkScheduler(serviceScopeFactory, loggerFactory);
+
+            using var executingSignal = new ManualResetEventSlim(false);
+            using var cancelledSignal = new ManualResetEventSlim(false);
+            using var executedSignal = new ManualResetEventSlim(false);
+
+            WorkItem workItem = (state, sp, ct) =>
+            {
+                executingSignal.Set();
+
+                cancelledSignal.Wait(100);
+                Assert.True(ct.IsCancellationRequested);
```

Comment:
As above, I think a false assertion would be caught by the `Execute()` try catch.